xmtp/src/builder.rs

```diff
@@ -33,14 +36,47 @@ where
         self
     }
 
+    pub fn find_or_create_account(mut self, wallet_address: String) -> Result<Self, String> {
+        let key = get_account_storage_key(wallet_address);
+        let persistence = self.persistence.as_mut().ok_or_else(|| {
+            "Persistence engine must be set before setting the account".to_string()
+        })?;
+
+        let existing = persistence.read(key.clone());
+        match existing {
+            Ok(Some(data)) => {
+                let data_string = std::str::from_utf8(&data).map_err(|e| format!("{}", e))?;
+                let account: VmacAccount =
+                    serde_json::from_str(data_string).map_err(|e| format!("{}", e))?;
+                self.account = Some(account)
+            }
+            Ok(None) => {
+                let account = VmacAccount::generate();
+                let data = serde_json::to_string(&account).map_err(|e| format!("{}", e))?;
+
+                persistence.write(key, data.as_bytes())?;
+
+                self.account = Some(account)
+            }
+            Err(e) => return Err(format!("Failed to read from persistence: {}", e)),
+        }
+
+        Ok(self)
+    }
+
```

Comment:
This seems like it should be on the client not the builder.

Using the builder pattern it shouldn't matter than order the params are built. and the result return type will make chaining need an unwrap. 

Following the pattern, the builder would accept a Option<wallet_address> and then the client would fetch and store.

Reply:
So, the reason I didn't want to put it on the client is because I didn't want the client to have an optional account field. Feels like it makes life easier to have as many fields as possible be required at the time of creation, so we don't have to have a bunch of option checks in the client code. 

But maybe the juice isn't worth the squeeze here.

Reply:
One alternative would be to have the `build()` method of the builder do the work to load the data. Then you still get a clean client.